examples/vnet/doca_gw_dpdk.c

```diff
@@ -1150,59 +1107,64 @@ doca_gw_dpdk_init_port(uint16_t port_id)
  *
  * @return 
  */
-static struct doca_gw_pipe_dpdk_flow*
-doca_gw_dpdk_create_pipe_flow(struct doca_gw_pipeline_cfg *cfg, struct doca_gw_error *err)
+static int
+doca_gw_dpdk_create_pipe_flow(struct doca_gw_pipe_dpdk_flow *flow, 
+							struct doca_gw_pipeline_cfg *cfg,
+							struct doca_gw_error *err)
 {
 	int ret;
-	struct doca_gw_pipe_dpdk_flow *pipe_flow;
 
-	pipe_flow = doca_gw_dpdk_get_free_pipe();
-	if (pipe_flow == NULL) {
-		err->type = DOCA_ERROR_NOMORE_PIPE_RESOURCE;
-		return NULL;
-	}
-	pipe_flow->port_id = (uint16_t)cfg->port->port_id;  
-	pipe_flow->attr.ingress = 1;
-	pipe_flow->attr.group = 1; // group 0 jump group 1
-	ret = doca_gw_dpdk_build_item(cfg->match, pipe_flow, err);
+	flow->port_id = (uint16_t)cfg->port->port_id;
+	flow->attr.ingress = 1;
+	flow->attr.group = 1; // group 0 jump group 1
+	ret = doca_gw_dpdk_build_item(cfg->match, flow, err);
 	if (ret) {
 		err->type = DOCA_ERROR_PIPE_BUILD_IMTE_ERROR;
-		goto free_pipe;
+		return -1;
 	}
-	ret = doca_gw_dpdk_build_action(cfg, pipe_flow);
+	ret = doca_gw_dpdk_build_action(cfg, flow);
 	if (ret) {
 		err->type = DOCA_ERROR_PIPE_BUILD_ACTION_ERROR;
-		goto free_pipe;
+		return -1;
 	}
-	doca_dump_rte_flow("create pipe:", pipe_flow->port_id, &pipe_flow->attr,
-		pipe_flow->items, pipe_flow->actions);
-	return pipe_flow;
-free_pipe:
-	doca_gw_dpdk_pipe_release(pipe_flow);
-	return NULL;
+	doca_dump_rte_flow("create pipe:", flow->port_id, &flow->attr,
```

Comment:
maybe you shoud have doca_log for dump packets so you can disable/enable without marking code

Reply:
todo...